### ISSUE TITLE
Shallow test renderer calls shouldComponentUpdate on forceUpdate

### ISSUE THREAD

**Issue by d4rky-pl** [facebook/react#11236]
There is no extra guard or clause in `enqueueForceUpdate` when calling `render` which means the `render` always calls `shouldComponentUpdate` which can block even forced re-render during tests.

This for example [breaks shallow rendering in Enzyme when using MobX and observables](https://github.com/airbnb/enzyme/issues/1267).

Minimal test case:

```javascript
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow';
const renderer = new ShallowRenderer();

class TestComponent extends React.Component {
  render() { return <span/> }
}

describe('forceUpdate', () => {
  it('does not call shouldComponentUpdate', () => {
    let called = 0
    TestComponent.prototype.shouldComponentUpdate = () => called += 1
    const component = renderer.render(<TestComponent/>);
    renderer._instance.forceUpdate()
    expect(called).toBe(0) // <- this throws because called === 1
  })
})
```

**Comment by gaearon** [comment]
React `16.1.0-beta` has been released. Please update `react`, `react-dom`, and `react-test-renderer` (if you use it) to this version and let us know if it solved the issue! We’d appreciate if you could test before Monday when we plan to get `16.1.0` out.

**Comment by d4rky-pl** [comment]
@gaearon Hey, it seems that I actually botched this fix - the shouldComponentUpdate() is no longer called but the component is still not rerendered properly (check was done in wrong `if` clause). I've sent another PR that resolves this, please check #11439.

Apologies for extra noise!
